boa_engine/src/vm/tests.rs

```diff
@@ -1,33 +1,38 @@
-use boa_parser::Source;
-
-use crate::{check_output, exec, Context, JsValue, TestAction};
+use crate::{run_test, JsValue, TestAction};
+use indoc::indoc;
 
 #[test]
 fn typeof_string() {
-    let typeof_object = r#"
-        const a = "hello";
-        typeof a;
-    "#;
-    assert_eq!(&exec(typeof_object), "\"string\"");
+    run_test([TestAction::assert_eq(
```

Comment:
Most of these tests are only passing 1 item in the array, do we want an API which can handle just a single test?
The name run_test doens't sound right when it's designed to take in an array of tests. But I understand this is just bikeshedding.

Reply:
`run_test_actions` maybe?

Reply:
I think that name would be describing better what is happening yeah. (Even though it is longer)

Reply:
> Most of these tests are only passing 1 item in the array, do we want an API which can handle just a single test?

Thought about it, but I didn't think it would be worth increasing the complexity of the API just to save two (`[` and `]`) keystrokes...

... but now that I think about it, we could implement `IntoIterator` for `TestAction`, which doesn't increase our API complexity, though it makes `run_test_actions(TestAction::exec(..))` look a bit weird.